src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT c.name as car_name, c.id as car_id, "
+                + "m.name as manufacturer_name, m.country as manufacturer_country, "
+                + "m.id as manufacturer_id  from cars_drivers cd "
+                + "LEFT JOIN cars c ON cd.car_id = c.id "
+                + "LEFT JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE driver_id = ? AND c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsByDriverStatement
+                        = connection.prepareStatement(getAllByDriverRequest)) {
+            getCarsByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get cars by drivers id " + driverId,
+                    throwable);
+        }
+        parseDriversToListOfCars(cars);
+        return cars;
+    }
+
+    private void removeDriversRelations(Car car) {
+        String deleteRequest = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeStatement = connection.prepareStatement(deleteRequest)) {
+            removeStatement.setLong(1, car.getId());
+            removeStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't delete drivers dependencies for car with id "
+                    + car.getId(), throwable);
+        }
+    }
+
+    private void addDriverRelationForCar(Car car, Driver driver) {
+        String insertDriverRelation
+                = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement
+                        = connection.prepareStatement(insertDriverRelation)) {
+            insertStatement.setLong(1, car.getId());
+            insertStatement.setLong(2, driver.getId());
+            insertStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't delete drivers dependencies for car with id "
+                    + car.getId(), throwable);
+        }
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getAllDriversForCarRequest = "SELECT d.id as driver_id, d.name as driver_name, "
+                + "d.license_number FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id WHERE cd.car_id = ?;";
```

Comment:
Let's add where condition to ignore deleted drivers

Reply:
done